src/components/Form/CustomTemplates/FieldTemplate.tsx

```diff
@@ -0,0 +1,83 @@
+import FormControl from '@mui/material/FormControl';
+import Typography from '@mui/material/Typography';
+import {
+	FieldTemplateProps,
+	FormContextType,
+	RJSFSchema,
+	StrictRJSFSchema,
+	getTemplate,
+	getUiOptions,
+} from '@rjsf/utils';
+
+export const FieldTemplate = <
+	T = any,
+	S extends StrictRJSFSchema = RJSFSchema,
+	F extends FormContextType = any,
+>(
+	props: FieldTemplateProps<T, S, F>,
+) => {
+	const {
+		id,
+		children,
+		classNames,
+		style,
+		disabled,
+		displayLabel,
+		hidden,
+		label,
+		onDropPropertyClick,
+		onKeyChange,
+		readonly,
+		required,
+		rawErrors = [],
+		errors,
+		help,
+		description,
+		rawDescription,
+		schema,
+		uiSchema,
+		registry,
+	} = props;
+	const uiOptions = getUiOptions<T, S, F>(uiSchema);
+	const WrapIfAdditionalTemplate = getTemplate<
+		'WrapIfAdditionalTemplate',
+		T,
+		S,
+		F
+	>('WrapIfAdditionalTemplate', registry, uiOptions);
+
+	if (hidden) {
+		return <div style={{ display: 'none' }}>{children}</div>;
+	}
+	return (
+		<WrapIfAdditionalTemplate
+			classNames={classNames}
+			style={style}
+			disabled={disabled}
+			id={id}
+			label={label}
+			onDropPropertyClick={onDropPropertyClick}
+			onKeyChange={onKeyChange}
+			readonly={readonly}
+			required={required}
+			schema={schema}
+			uiSchema={uiSchema}
+			registry={registry}
+		>
+			{displayLabel && rawDescription ? (
+				<Typography variant="caption" color="textSecondary">
+					{description}
+				</Typography>
+			) : null}
+			<FormControl
+				fullWidth={true}
+				error={rawErrors.length ? true : false}
+				required={required}
+			>
+				{children}
+				{errors}
+				{help}
+			</FormControl>
```

Comment:
What happens if we have both a description and a label for the input?

Reply:
This comment made me think that I'm actually wrong, I should not fix this here as the description cannot stay between the label and the input. The point is that since the label is the only html element that has a htmlFor attribute and it is also the only elements that should interact with the input, we should leave it as it is. 

We should apply the fix in the configVar schema instead which return something like this:

```
default: 16
description: "Define device GPU memory in megabytes."
type: "integer"
will_reboot: true
```

If we want the description to be the label of the input then we need to change description to label

Reply:
@myarmolinsky after a long thinking on it, I would actually go with this https://github.com/balena-io/balena-ui/pull/6188